cmd/cloudinitsave/cloudinitsave.go

```diff
@@ -46,73 +48,41 @@ const (
 	datasourceInterval    = 100 * time.Millisecond
 	datasourceMaxInterval = 30 * time.Second
 	datasourceTimeout     = 5 * time.Minute
+	configDev             = "LABEL=config-2"
+	configDevMountPoint   = "/media/config-2"
 )
 
-var (
-	network bool
-	flags   *flag.FlagSet
-)
-
-func init() {
-	flags = flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
-	flags.BoolVar(&network, "network", true, "use network based datasources")
-}
-
 func Main() {
-	flags.Parse(os.Args[1:])
+	log.Info("Running cloud-init-save")
 
-	log.Infof("Running cloud-init-save: network=%v", network)
+	cfg := rancherConfig.LoadConfig()
+	network.ApplyNetworkConfig(cfg)
 
-	if err := saveCloudConfig(); err != nil {
+	if err := SaveCloudConfig(true); err != nil {
 		log.Errorf("Failed to save cloud-config: %v", err)
 	}
 }
 
-func saveFiles(cloudConfigBytes, scriptBytes []byte, metadata datasource.Metadata) error {
-	os.MkdirAll(rancherConfig.CloudConfigDir, os.ModeDir|0600)
-
-	if len(scriptBytes) > 0 {
-		log.Infof("Writing to %s", rancherConfig.CloudConfigScriptFile)
-		if err := util.WriteFileAtomic(rancherConfig.CloudConfigScriptFile, scriptBytes, 500); err != nil {
-			log.Errorf("Error while writing file %s: %v", rancherConfig.CloudConfigScriptFile, err)
-			return err
-		}
-	}
-
-	if len(cloudConfigBytes) > 0 {
-		if err := util.WriteFileAtomic(rancherConfig.CloudConfigBootFile, cloudConfigBytes, 400); err != nil {
-			return err
-		}
-		log.Infof("Written to %s:\n%s", rancherConfig.CloudConfigBootFile, string(cloudConfigBytes))
-	}
-
-	metaDataBytes, err := yaml.Marshal(metadata)
-	if err != nil {
+func MountConfigDrive() error {
+	if err := os.MkdirAll(configDevMountPoint, 644); err != nil {
 		return err
 	}
 
-	if err = util.WriteFileAtomic(rancherConfig.MetaDataFile, metaDataBytes, 400); err != nil {
-		return err
+	configDev := util.ResolveDevice(configDev)
+
+	if configDev == "" {
+		return mount.Mount("config-2", configDevMountPoint, "9p", "trans=virtio,version=9p2000.L")
```

Comment:
Is this comment saying the same thing as your first comment? I'm not sure what you mean.